### PR TITLE
Keep Adam bias correction in-graph so optimizer steps can be jit-compiled

### PR DESCRIPTION
**Pull request by mlajtos**
## Problem

`treeBiasCorrection` reads the step count with `count.item()`:

```ts
const correction = 1 / (1 - Math.pow(decay, count.item()));
```

`.item()` is a data read, so it throws on tracers — which means any `update()` built on bias correction (`adam`, `adamw`, …) dies at trace time inside `jit()`. A jit-compiled training step is only possible with `sgd`; Adam steps silently force users back to eager, op-by-op dispatch.

## Fix

Compute the correction with tensor ops — the count stays in the graph:

```ts
const correction = np.reciprocal(
  np.subtract(1.0, np.power(decay, count.astype(np.float32))),
);
```

Numerically identical up to float32 ordering (the old path computed the scalar in float64 and multiplied; early Adam steps divide by near-zero denominators like `1 − 0.999¹`, so low digits legitimately differ between orderings).

The other `count.item()` sites (`scaleBySchedule`, schedule-valued weight decay) are untouched — they feed JS callbacks and genuinely need the value.

## Test

Adds `adamw steps can be jit-compiled and match eager`: three jit-compiled `adamw` steps run (previously: throws at trace time) and match three eager steps within float32 tolerance (`toBeAllclose`, rtol 1e-2).

`pnpm exec vitest run packages/optax`: 32/32 pass. `pnpm check` and `pnpm lint --max-warnings 0` clean.

### CODE REVIEW

**Review by ekzhang — APPROVED** [review]
Thank you!
